src/celeritas/phys/Interaction.hh

```diff
@@ -76,10 +76,38 @@ struct MscStep
     real_type phys_step{};        //!< Step length from physics processes
     real_type true_path{};        //!< True path length due to the msc
     real_type geom_path{};        //!< Geometrical path length
-    real_type limit_min{};        //!< Minimum of the true path limit
     real_type alpha{-1};          //!< An effecive mfp rate by distance
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Persistent range properties for multiple scattering (msc) within a volume.
+ *
+ * These values are calculated at the first step in every msc tracking volume
+ * and reused at subsequent steps within the same volume.
+ */
+struct MscRange
+{
+    real_type range_init{}; //!< Initial msc range
+    real_type range_fact{}; //!< Scale factor for the msc range
+    real_type limit_min{};  //!< Minimum of the true path limit
+
+    CELER_FUNCTION bool is_filled() const
```

Comment:
Maybe make this `explicit operator bool`?

Reply:
Replaced!